src/material/slider/slider.ts

```diff
@@ -659,671 +1247,584 @@ export class MatSlider
+        left: `${sibling.translateX - this._rippleRadius}px`,
+        right: 'auto',
+        transformOrigin: 'left',
+        transform: `scaleX(${activePercentage})`,
+      });
     }
   }
 
-  /**
-   * Returns the thumb position of the given event target.
-   * Returns null if the given event target does not correspond to a slider thumb input.
-   */
-  private _getInputThumbPosition(target: EventTarget | null): Thumb | null {
-    if (target === this._delegate._getInputElement(Thumb.END)) {
-      return Thumb.END;
+  private _updateTrackUINonRange(source: MatSliderThumb): void {
+    this._isRtl
+      ? this._setTrackActiveStyles({
+          left: 'auto',
+          right: '0px',
+          transformOrigin: 'right',
+          transform: `scaleX(${1 - source.fillPercentage})`,
+        })
+      : this._setTrackActiveStyles({
+          left: '0px',
+          right: 'auto',
+          transformOrigin: 'left',
+          transform: `scaleX(${source.fillPercentage})`,
+        });
+  }
+
+  // Tick mark update conditions
+  //
+  // 1. Value
+  //    - Reason: a tick mark which was once active might now be inactive or vice versa.
+  // 2. Min, max, or step
+  //    - Reason #1: the number of tick marks may have changed.
+  //    - Reason #2: The value may have silently changed.
+
+  /** Updates the dots along the slider track. */
+  _updateTickMarkUI(): void {
+    if (this.step === undefined || this.min === undefined || this.max === undefined) {
+      return;
     }
-    if (this._delegate._isRange() && target === this._delegate._getInputElement(Thumb.START)) {
-      return Thumb.START;
+    const step = this.step > 0 ? this.step : 1;
+    this._isRange ? this._updateTickMarkUIRange(step) : this._updateTickMarkUINonRange(step);
+
+    if (this._isRtl) {
+      this._tickMarks.reverse();
     }
-    return null;
+    this._cdr.markForCheck();
   }
 
-  // We manually assign functions instead of using prototype methods because
-  // MDC clobbers the values otherwise.
-  // See https://github.com/material-components/material-components-web/pull/6256
+  private _updateTickMarkUINonRange(step: number): void {
+    const value = this._getValue();
+    let numActive = Math.floor((value - this.min) / step);
+    let numInactive = Math.floor((this.max - value) / step);
+    this._isRtl ? numActive++ : numInactive++;
+
+    this._tickMarks = Array.from({length: numActive})
+      .map(() => TickMark.ACTIVE)
+      .concat(Array.from({length: numInactive}).map(() => TickMark.INACTIVE));
+  }
+
+  private _updateTickMarkUIRange(step: number): void {
+    const endValue = this._getValue();
+    const startValue = this._getValue(Thumb.START);
+    const numInactiveBeforeStartThumb = Math.floor((startValue - this.min) / step);
+    const numActive = Math.floor((endValue - startValue) / step) + 1;
+    const numInactiveAfterEndThumb = Math.floor((this.max - endValue) / step);
+    this._tickMarks = Array.from({length: numInactiveBeforeStartThumb})
```

Comment:
You can simplify these `Array.from().map()` calls to `Array(length).fill(value)`.

Reply:
or alternatively `Array.from` takes a second argument which is a mapping function